### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Dangelo-JAN**
# I create my first integrations

In this milestone I:

-  Use Capybara to write integration tests for each view in my project.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post-show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each comment.
  - I can see the comment each comment left.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = Post.all
+    @posts = Post.where(author_id: @user)
   end
```
Comment (TedLivist) [inline app/controllers/posts_controller.rb]: - [x] You are yet to eliminate the N + 1 query problem for the Post index. You can do that by refactoring to this:

  ```
  def index
    @user = User.find(params[:user_id])
    @posts = @user.posts.includes(:comments)
  end
  ```

File: spec/integration/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+RSpec.describe 'Post index', type: :requests do
```
Comment (TedLivist) [inline spec/integration/post_index_spec.rb:3]: - [x] For the Post index page test, you seem to have skipped the following test scenarios. Kindly add them:
  
  - Tests that asserts you can see some of the post's body.
  - Tests that redirects to the post's show page when you click on a post.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Dangelo-JAN and @alexpuente0,

Good job so far!
✔️ Good documentation
✔️ Available tests are passing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Dangelo-JAN) [reply app/controllers/posts_controller.rb]: done

File: spec/integration/post_index_spec.rb (hunk 2 above)
Comment (Dangelo-JAN) [reply spec/integration/post_index_spec.rb:3]: done

**Review by Whoistolu — APPROVED** [review]
### Hi @Dangelo-JAN 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**